### DISCUSSION TITLE
Raspberry Pi 3 B+ server crashes

### DISCUSSION THREAD

**Discussion by FraxinusErit** [Pryaxis/TShock#1967] [Q&A] [answered]
Hello everyone

I am running the newest version of Tshock on my raspberry pi, and it works really smooth, but it crashes on occasion. I thought that perhaps this was due to lots of players being active, but it does the same thing with single player, and occasionally it crashes when nobody is logged in. I thought that perhaps the issue was cooling so I hooked up two fans onto the pi, but I still have problems. It seems like the server tends to crash every so often when it auto saves. What might be my problem, how do I fix it, and can a raspberry pi really run a Terraria server at all? I was reading a bunch of sites and people said that they had no problem with it, but I think journey's end might be too much for the pi to handle.

thanks!

**Comment by hakusaro** [comment, marked as answer]
You could try the [tiled plugin by thanatos-tshock](https://github.com/pryaxis/plugins), with a more optimized tile implementation.

We don't really support the Pi as a destination platform. It does work but it's not ideal. I know a lot of other RPi users exist though. I think one of the tricks is tiled.

**Reply by FraxinusErit** [reply]
Is the current version new enough to work? My server is in 1.4.0.4

**Reply by hakusaro** [reply]
I'm not personally sure what the update status looks like, but you can always try it and turn it off it off.

**Reply by hakusaro** [reply]
@thanatos-tshock do you know?

**Reply by FraxinusErit** [reply]
no I don't, but my guess is that it might work just because it operates on the basic principle of tiles, but I'm not sure. I will give it a shot.

**Reply by FraxinusErit** [reply]
I installed it and it worked beautifully! the server is running well now, but the only issue is the current desyncing issue with Npcs, especially bosses.

**Reply by hakusaro** [reply]
That's probably going to be a CPU bound issue :( we've confirmed that having a low power CPU results in a lot of work for the NPC sync stuff.

**Reply by FraxinusErit** [reply]
really? The vanilla server worked when it came to desyncing, but it also crashed because of the tile loader. Is there a way to replace the tile loading software on the vanilla version (I don't exactly know what I'm talking about so if that's completely stupid sorry).

**Reply by hakusaro** [reply]
> Is there a way to replace the tile loading software on the vanilla version (I don't exactly know what I'm talking about so if that's completely stupid sorry)

nada

The NPC issue is a vanilla issue -- we worked with Re-Logic extensively on it.

**Reply by FraxinusErit** [reply]
Ok, after I saw your message I went back to the vanilla server type. There were some desyncs, but they weren't quite as major. We weren't having desyncing issues before 1.4.0.3 as far as I remember. So when you say it's a vanilla issue, is it something that future updates will fix?

**Reply by FraxinusErit** [reply]
I have been getting this eror message, and I thought I would send it in case it is important.

UpdateManager warning: Error: SecureChannelFailure (the authentication or decryption has failed)
-> the authentication or decryption has failed
-> the authentication or decryption has failed
-> the authentication or decryption has failed
